### PR TITLE
Intro Sequence Or Something Idk What To Call It

### PR DESCRIPTION
**Pull request by tyeagg**
## About The Pull Request
its like in the movies!!
op names list from cm

https://user-images.githubusercontent.com/98577688/194579795-153a6b47-bf7b-4f60-a887-1dbfd50f2a07.mp4

image

stuff that plays when player enters ground map in combat patrol:
image
image

## Why It's Good For The Game
fluff good 

## Changelog
:cl:
add: intro sequence thing in first drop
imageadd: lrprr, tdf, black ops, shokk emblems for deployment intro sequence and portraits of som and tgmc overwatch
/:cl:

### CODE REVIEW

**Comment by Weedmaster77** [comment]
cool idea

**Comment by MLGTASTICa** [comment]
what if captain could set operation name?

**Comment by XSlayer300** [comment]
For Crash and Combat Patrol:
- Crash: add the intro sequence when the marines start to land the shuttle down.
- Combat Patrol: add the intro sequence when the gates opened up.

**Comment by bowlofcereal** [comment]
Give the Captain something like a captain's charter where they can name the operation, and its name will appear on the BYOND Hub if possible, like station names. This is a really cool concept I didnt know we needed.

For maximum stupidity you should give the text that scrawls sound effects, like for example the arma sfx or something for that kind of fluff.

**Comment by novaepee** [comment]
Tivi says no to changing operation name:

image

File: code/__DEFINES/layers.dm [hunk 1]
```diff
@@ -223,6 +223,8 @@
 #define HUD_LAYER 24
 #define ABOVE_HUD_PLANE 2000
 #define ABOVE_HUD_LAYER 25
+#define INTRO_PLANE 2001
```
Comment (TiviPlus) [inline code/__DEFINES/layers.dm:226]: Omgg!!!! just like the movie!!
2001 a space odyssey!!!!

File: code/modules/shuttle/marine_dropship.dm [hunk 2]
```diff
@@ -167,6 +169,12 @@
 		return
 	// pull the shuttle from datum/source, and state info from the shuttle itself
 	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_DROPSHIP_TRANSIT)
+	if(first_landing)
+		for(var/mob/living/carbon/human/human AS in GLOB.alive_human_list)
```
Comment (TiviPlus) [inline code/modules/shuttle/marine_dropship.dm:173]: Optional:
Check for gamemode here? will stop any funnies if an admin moves the shuttle for an event or whatever

File: code/datums/gamemodes/combat_patrol.dm [hunk 3]
```diff
@@ -81,8 +81,17 @@
 	var/datum/game_mode/combat_patrol/D = SSticker.mode
 	addtimer(CALLBACK(D, /datum/game_mode/combat_patrol.proc/set_game_timer), SSticker.round_start_time + shutters_drop_time + 5 MINUTES) //game cannot end until at least 5 minutes after shutter drop
 	addtimer(CALLBACK(D, /datum/game_mode/combat_patrol.proc/respawn_wave), SSticker.round_start_time + shutters_drop_time) //starts wave respawn on shutter drop and begins timer
+	addtimer(CALLBACK(D, /datum/game_mode/combat_patrol.proc/intro_sequence), SSticker.round_start_time + shutters_drop_time - 10 SECONDS) //starts intro sequence 10 seconds before shutter drop
 	TIMER_COOLDOWN_START(src, COOLDOWN_BIOSCAN, SSticker.round_start_time + shutters_drop_time + bioscan_interval)
 
+///intro sequence
```
Comment (TiviPlus) [inline code/datums/gamemodes/combat_patrol.dm:87]:
```suggestion
/// plays the intro sequence
```

File: code/datums/gamemodes/combat_patrol.dm [hunk 4]
```diff
@@ -81,8 +81,17 @@
...
+/datum/game_mode/combat_patrol/proc/intro_sequence()
+	for(var/mob/living/carbon/human/human AS in GLOB.alive_human_list)
+		if(human.faction == FACTION_TERRAGOV)
+			human.play_screen_text("<span class='maptext' style=font-size:24pt;text-align:left valign='top'><u>[GLOB.operation_namepool[/datum/operation_namepool].get_random_name()]</u></span><br>" + "[SSmapping.configs[GROUND_MAP].map_name]<br>" + "[GAME_YEAR]-[time2text(world.realtime, "MM-DD")] [stationTimestamp("hh:mm")]<br>" + "Territorial Defense Force Platoon<br>" + "[human.job.title], [human]<br>", /obj/screen/text/screen_text/command_order/intro)
+		else
+			human.play_screen_text("<span class='maptext' style=font-size:24pt;text-align:left valign='top'><u>[GLOB.operation_namepool[/datum/operation_namepool].get_random_name()]</u></span><br>" + "[SSmapping.configs[GROUND_MAP].map_name]<br>" + "[GAME_YEAR]-[time2text(world.realtime, "MM-DD")] [stationTimestamp("hh:mm")]<br>" + "Shokk Infantry Platoon<br>" + "[human.job.title], [human]<br>", /obj/screen/text/screen_text/command_order/intro)
```
Comment (TiviPlus) [inline code/datums/gamemodes/combat_patrol.dm:93]:
```suggestion
			human.play_screen_text("<span class='maptext' style=font-size:24pt;text-align:left valign='top'><u>[GLOB.operation_namepool[/datum/operation_namepool].get_random_name()]</u></span><br>" + "[SSmapping.configs[GROUND_MAP].map_name]<br>" + "[GAME_YEAR]-[time2text(world.realtime, "MM-DD")] [stationTimestamp("hh:mm")]<br>" + "Shock Infantry Platoon<br>" + "[human.job.title], [human]<br>", /obj/screen/text/screen_text/command_order/intro)
```

File: code/datums/namepool.dm [hunk 5]
```diff
@@ -71,3 +72,15 @@ GLOBAL_LIST_EMPTY_TYPED(namepool, /datum/namepool)
 /datum/namepool/robotic/get_random_name(gender = MALE)
 	. = pick(SSstrings.get_list_from_file(firstname_female_pool))
 	. += "-[rand(1,999)]" //pathfinder-738 or such
+
+/datum/operation_namepool
+	var/list/operation_titles = "names/operation_title"
+	var/list/operation_prefixes = "names/operation_prefix"
+	var/list/operation_postfixes = "names/operation_postfix"
```
Comment (TiviPlus) [inline code/datums/namepool.dm:79]: autodoc

File: code/datums/namepool.dm [hunk 6]
```diff
@@ -71,3 +72,15 @@ GLOBAL_LIST_EMPTY_TYPED(namepool, /datum/namepool)
 /datum/namepool/robotic/get_random_name(gender = MALE)
 	. = pick(SSstrings.get_list_from_file(firstname_female_pool))
 	. += "-[rand(1,999)]" //pathfinder-738 or such
+
+/datum/operation_namepool
+	var/list/operation_titles = "names/operation_title"
+	var/list/operation_prefixes = "names/operation_prefix"
```
Comment (TiviPlus) [inline code/datums/namepool.dm:78]: Suggestion for after you finish this pr, you can remove this and make the operation name like tg's station names (see /proc/new_station_name())

File: code/datums/namepool.dm [hunk 7]
```diff
@@ -71,3 +72,15 @@ GLOBAL_LIST_EMPTY_TYPED(namepool, /datum/namepool)
...
+	var/list/operation_postfixes = "names/operation_postfix"
+	var/operation_name
+
+/datum/operation_namepool/proc/get_random_name()
+	operation_name = pick(SSstrings.get_list_from_file(operation_titles))
+	operation_name += " [pick(SSstrings.get_list_from_file(operation_prefixes))]"
+	operation_name += "-[pick(SSstrings.get_list_from_file(operation_postfixes))]"
+	return uppertext(operation_name)
```
Comment (TiviPlus) [inline code/datums/namepool.dm:86]:
```suggestion

/datum/operation_namepool/proc/get_random_name()
	var/operation_name = pick(SSstrings.get_list_from_file(operation_titles))
	operation_name += " [pick(SSstrings.get_list_from_file(operation_prefixes))]"
	operation_name += "-[pick(SSstrings.get_list_from_file(operation_postfixes))]"
	return uppertext(operation_name)
```

File: code/datums/gamemodes/combat_patrol.dm (hunk 4 above)
Comment (tyeagg) [reply code/datums/gamemodes/combat_patrol.dm:93]: but its cool like som language is different than terragov language idk

**Comment by comfyorange** [comment]
#### Test Merge Deployed

##### Server Instance
TGMC

##### Merged By
GAME\naaanii

##### Revision
Origin: d9f05eb44f958814783b33f5e725a88de2ce56cb
Pull Request: 279c9fa706e78b7cc965b79e159e131fb8c641de
Server: a1f8f0c0a902ff188116e56b2f95f1572f222a74
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
GAME\tiviplus

##### Revision
Origin: f672f018d98fbc3542ada38d6260f37d4c50d0e3
Pull Request: 279c9fa706e78b7cc965b79e159e131fb8c641de
Server: 0f2e13d46370ed4b71632488688e40bdd3acef2e

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Removed

**Comment by FosterGrime-Heymynameslouie** [comment]
Screenshot_20220930_140231

too lahrged or might be a problem on my end?

**Comment by comfyorange** [comment]
#### Test Merge Deployed

##### Server Instance
TGMC

##### Merged By
GAME\tiviplus

##### Revision
Origin: 3398322f1f506d5ac8ae36ca492d3fb015506b02
Pull Request: 64fe66dd5f3ffbb73de9eedc1bb8d29d9d033c96
Server: 8a3c8b025777ba377bb061b61a5c7ec23f61e9af

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by bowlofcereal** [comment]
bugged, plays every time the dropship cycles with a new random name

**Comment by tyeagg** [comment]
i honestly dont know why its so inconsistent because sometimes its not bugged and sometimes it is

File: code/modules/shuttle/marine_dropship.dm [hunk 8]
```diff
@@ -167,6 +169,12 @@
...
+			if(human.faction != FACTION_TERRAGOV)
+				return
+			human.play_screen_text("<span class='maptext' style=font-size:24pt;text-align:left valign='top'><u>[GLOB.operation_namepool[/datum/operation_namepool].get_random_name()]</u></span><br>" + "[SSmapping.configs[GROUND_MAP].map_name]<br>" + "[GAME_YEAR]-[time2text(world.realtime, "MM-DD")] [stationTimestamp("hh:mm")]<br>" + "Long Range Patrol Rapid Response Platoon<br>" + "[human.job.title], [human]", /obj/screen/text/screen_text/command_order/intro)
+		first_landing = FALSE
```
Comment (XSlayer300) [inline code/modules/shuttle/marine_dropship.dm:177]:
```suggestion
		first_landing = FALSE
```
Put first_landing first, before play_screen_text?

**Comment by XSlayer300** [comment]
So just to note here. Since the Tadpole and CAS counts as a 'dropship', it inadvertently the intro text out for some reason. Make it restricted only to the deployment ship.

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
GAME\tiviplus

##### Revision
Origin: 97bb57b1283bcd96910a866f9b257d4422f807e6
Pull Request: 64fe66dd5f3ffbb73de9eedc1bb8d29d9d033c96
Server: 55af5738f09e4f32af093c8b6b6aba41f6390460

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: c1d51908261666015798601def9aec4333609250
Pull Request: 64fe66dd5f3ffbb73de9eedc1bb8d29d9d033c96
Server: 568d2aff76b2c7ed883b6e6dfeedd8de00c4c4b1

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
GAME\tiviplus

##### Revision
Origin: 4836928d30d2a5e28694d9c1ab705e56bfe4a3f0
Pull Request: 64fe66dd5f3ffbb73de9eedc1bb8d29d9d033c96
Server: 1d2c6c0233f162c9f43a26214749e40edf56ebe6

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
GAME\tiviplus

##### Revision
Origin: 2968255260c476ffe21a73032b3618818c8c9a3a
Pull Request: 64fe66dd5f3ffbb73de9eedc1bb8d29d9d033c96
Server: 6c70c2d724f76aebfcc1e0d62b88b004f74c4359

##### Comment
No comment set - Fast Add

File: code/modules/shuttle/gamemodes/crash.dm [hunk 9]
```diff
@@ -46,6 +46,13 @@
 	. = ..()
 	SSshuttle.canterbury = src
 
+/obj/docking_port/mobile/crashmode/on_prearrival()
+	. = ..()
+	for(var/mob/living/carbon/human/human AS in GLOB.alive_human_list)
+		if(human.faction != FACTION_TERRAGOV)
+			return
+		human.play_screen_text("<span class='maptext' style=font-size:24pt;text-align:left valign='top'><u>[GLOB.operation_namepool[/datum/operation_namepool].get_random_name()]</u></span><br>" + "[SSmapping.configs[GROUND_MAP].map_name]<br>" + "[GAME_YEAR]-[time2text(world.realtime, "MM-DD")] REDACTED<br>" + "Black Ops Platoon<br>" + "[human.job.title], REDACTED", /obj/screen/text/screen_text/command_order/intro/blackop)
```
Comment (XSlayer300) [inline code/modules/shuttle/gamemodes/crash.dm:54]:
```suggestion
		human.play_screen_text("<span class='maptext' style=font-size:24pt;text-align:left valign='top'><u>[GLOB.operation_namepool[/datum/operation_namepool].get_random_name()]</u></span><br>" + "[SSmapping.configs[GROUND_MAP].map_name]<br>" + "[GAME_YEAR]-[time2text(world.realtime, "MM-DD")] [REDACTED]<br>" + "Black Ops Platoon<br>" + "[human.job.title], [REDACTED]", /obj/screen/text/screen_text/command_order/intro/blackop)
```

File: code/modules/shuttle/gamemodes/crash.dm [hunk 10]
```diff
@@ -46,6 +46,13 @@
 	. = ..()
 	SSshuttle.canterbury = src
 
+/obj/docking_port/mobile/crashmode/on_prearrival()
+	. = ..()
+	for(var/mob/living/carbon/human/human AS in GLOB.alive_human_list)
+		if(human.faction != FACTION_TERRAGOV)
+			return
+		human.play_screen_text("<span class='maptext' style=font-size:24pt;text-align:left valign='top'><u>[GLOB.operation_namepool[/datum/operation_namepool].get_random_name()]</u></span><br>" + "[SSmapping.configs[GROUND_MAP].map_name]<br>" + "[GAME_YEAR]-[time2text(world.realtime, "MM-DD")] █:█<br>" + "Black Ops Platoon<br>" + "[human.job.title], ████", /obj/screen/text/screen_text/command_order/intro/blackop)
```
Comment (XSlayer300) [inline code/modules/shuttle/gamemodes/crash.dm:54]: Was about to suggest that, thanks.
image

**Review by Kuro020 — APPROVED** [review]

**Comment by FosterGrime-Heymynameslouie** [comment]
this is so cool how is this even remotely possibel

speedmerhg

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 6aea506e5d51321aff9be738b1801a71deaa7673
Pull Request: 9ae062aab67f765ec7668b1e9a5faf138b46ef6b
Server: ce3040c5092889586ff1b0d7b9932cee5804dc07

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

File: code/modules/screen_alert/command_alert.dm [hunk 11]
```diff
@@ -12,6 +12,44 @@
 	style_open = "<span class='maptext' style=font-size:20pt;text-align:center valign='top'>"
 	style_close = "</span>"
 
+/obj/screen/text/screen_text/command_order/intro
```
Comment (TiviPlus) [inline code/modules/screen_alert/command_alert.dm:15]: 1) dont get why this is a command_order child
2) also should probably be renamed from /intro to something like /picture so it's clearer if/when we use it for something new

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest lgtm I think if you're happy with this

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
GAME\tiviplus

##### Revision
Origin: 61cd35e853782933f17fcb554a886c92e89910bf
Pull Request: b06ecc9b134a6bf2a3020e0c9bee75ccdde47953
Server: d3b601f73594225f09296266aeca424ba5f63e47

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
GAME\tiviplus

##### Revision
Origin: 61cd35e853782933f17fcb554a886c92e89910bf
Pull Request: b06ecc9b134a6bf2a3020e0c9bee75ccdde47953
Server: 2c4100570dc320458316804b137db11d4976d03b

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
GAME\tiviplus

##### Revision
Origin: 61cd35e853782933f17fcb554a886c92e89910bf
Pull Request: b06ecc9b134a6bf2a3020e0c9bee75ccdde47953
Server: 9766cfa7e7d99647d4c82b590054102718700d07

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by tyeagg** [comment]
yea im happy enough with this feel free to merge after i do the requested changes

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 0449553ec50b668e84c953056ff5b1182490c93a
Pull Request: b06ecc9b134a6bf2a3020e0c9bee75ccdde47953
Server: 1630e4f3365b2c34217380f6bfd89e5afb2b0b9f

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 3bb051ba922305abb667b07f59e39b5e379de56f
Pull Request: b06ecc9b134a6bf2a3020e0c9bee75ccdde47953
Server: 55bfb11f0f1099f61b533ad765a20c3bd148a7ea

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 3bb051ba922305abb667b07f59e39b5e379de56f
Pull Request: b06ecc9b134a6bf2a3020e0c9bee75ccdde47953
Server: 82be73307205f3692e8657c7c0651ae6bc1f3476

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 325a4cd17fc01492af3d6d99410839b754a648cd
Pull Request: b06ecc9b134a6bf2a3020e0c9bee75ccdde47953
Server: 04261d579813ea522558682b4efd03d3cbe978a1

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 15f80c44194c0d5d7aae462b24a4616f52467016
Pull Request: b06ecc9b134a6bf2a3020e0c9bee75ccdde47953
Server: ddcbfd545757618923a93d399a11668ea11b8169

##### Comment
No comment set - Fast Add
[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

**Comment by comfyorange** [comment]
#### Test Merge Removed

**Pull request merged**